src/graph/executor/query/FilterExecutor.cpp

```diff
@@ -84,6 +84,10 @@ Status FilterExecutor::handleSingleJobFilter() {
   bool canMoveData = movable(inputVar);
   Result result = ectx_->getResult(inputVar);
   auto *iter = result.iterRef();
+  // Always reuse getNeighbors's dataset to avoid some go statement execution plan related issues
+  if (iter->isGetNeighborsIter()) {
+    canMoveData = true;
```

Comment:
Why add this?

Reply:
Reuse getNeighbors's dataset. Go statements do not cause write-read conflicts.